### PR TITLE
`azurerm_virtual_desktop_host_pool` - support for `public_network_access`

### LINKED ISSUE #22429: Support Host Pool properties.publicNetworkAccess
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Terraform Version

1.4.6

### AzureRM Provider Version

3.64.0

### Affected Resource(s)/Data Source(s)

azurerm_virtual_desktop_host_pool

### Terraform Configuration Files

```hcl
I can provide if needed
```

### Debug Output/Panic Output

```shell
None
```

### Expected Behaviour

N/A

### Actual Behaviour

N/A

### Steps to Reproduce

N/A

### Important Factoids

_No response_

### References

The support of private networking is still in 'preview' it appears and if I remember correctly, HC doesn't touch these things until they are public 'no longer in preview'

If so then. great.

### PR DESCRIPTION
**Pull request by ASHR4**
## Community Note

* Please vote on this PR by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original PR to help the community and maintainers prioritize for review
* Please do not leave "+1" or "me too" comments, they generate extra noise for PR followers and do not help prioritize for review

## PR Checklist

- [x] I have followed the guidelines in our [Contributing Documentation](../blob/main/contributing/README.md).
- [x] I have checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change.
- [x] I have checked if my changes close any open issues. If so please include appropriate [closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) below.
- [x] I have updated/added Documentation as required written in a helpful and kind way to assist users that may be unfamiliar with the resource / data source.
- [x] I have used a meaningful PR title to help maintainers and other users understand this change and help prevent duplicate work. 
For example: “`resource_name_here` - description of change e.g. adding property `new_property_name_here`”

## Changes to existing Resource / Data Source

- [x] I have added an explanation of what my changes do and why I'd like you to include them (This may be covered by linking to an issue above, but may benefit from additional explanation).
- [x] I have written new tests for my resource or datasource changes & updated any relevent documentation.
- [x] I have successfully run tests with my changes locally. If not, please provide details on testing challenges that prevented you running the tests.
- [x] (For changes that include a **state migration only**). I have manually tested the migration path between relevant versions of the provider.

## Testing 

- [X] My submission includes Test coverage as described in the [Contribution Guide](../blob/main/contributing/topics/guide-new-resource.md) and the tests pass. (if this is not possible for any reason, please include details of why you did or could not add test coverage)

image

## Change Log

Below please provide what should go into the changelog (if anything) conforming to the [Changelog Format documented here](../blob/main/contributing/topics/maintainer-changelog.md).

* `azurerm_virtual_desktop_host_pool` - support for the `public_network_access_enabled` property 

This is a (please select all that apply):

- [ ] Bug Fix
- [ ] New Feature (ie adding a service, resource, or data source)
- [X] Enhancement
- [ ] Breaking Change

## Related Issue(s)
Fixes #22429

> [!NOTE] 
> If this PR changes meaningfully during the course of review please update the title and description as required.

### CODE REVIEW

File: internal/services/desktopvirtualization/virtual_desktop_host_pool_resource.go [hunk 1]
```diff
@@ -114,6 +115,12 @@ func resourceVirtualDesktopHostPool() *pluginsdk.Resource {
 				}, false),
 			},
 
+			"public_network_access_enabled": {
+				Type:     pluginsdk.TypeBool,
+				Optional: true,
+				Default:  true,
+			},
```
Comment (stephybun) [inline internal/services/desktopvirtualization/virtual_desktop_host_pool_resource.go:122]: There appear to be more values supported here than just enabled/disabled.

Although convention in the provider so far has been to expose this as a boolean, due to the rollout of additional network security features we need to begin exposing this as a string property.

Can we please change this to 
```suggestion
			"public_network_access": {
				Type:         pluginsdk.TypeString,
				Optional:     true,
				Default:      hostpool.HostpoolPublicNetworkAccessEnabled,
				ValidateFunc: validation.StringInSlice(hostpool.PossibleValuesForHostpoolPublicNetworkAccess(), false),
			},
```

**Review by stephybun — CHANGES_REQUESTED** [review]
Thanks for this PR @ASHR4. Would you mind taking a look over the suggestion left-line. In addition could you also confirm that the default value returned by the API if the field isn't specified at all is indeed `Enabled`? Once that's done we can continue with the review.

**Comment by ASHR4** [comment]
image

**Comment by ASHR4** [comment]
> Thanks for this PR @ASHR4. Would you mind taking a look over the suggestion left-line. In addition could you also confirm that the default value returned by the API if the field isn't specified at all is indeed `Enabled`? Once that's done we can continue with the review.

Hi @stephybun, if the field is not supplied it is indeed Enabled :)

**Comment by ASHR4** [comment]
Updated acceptance tests

image

**Review by stephybun — APPROVED** [review]
Thanks @ASHR4 LGTM 🦀

**Pull request merged**